jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/audit/ServicesAwareAuditEventBuilderTest.java

```diff
@@ -0,0 +1,158 @@
+        super.setUp();
+        processMetadata.put("CorrelationKey", correlationKey);
+
+        setUpMocks();
+    }
+
+    private void setUpMocks() {
+        when(kieRuntime.getIdentifier()).thenReturn(2L);
+
+        when(processInstance.getId()).thenReturn(1L);
+        when(processInstance.getDescription()).thenReturn("Some test Process");
+        when(processInstance.getSlaCompliance()).thenReturn(0);
+        when(processInstance.getSlaDueDate()).thenReturn(null);
+        when(processInstance.getMetaData()).thenReturn(processMetadata);
+
+        when(processInstance.getProcess()).thenReturn(process);
+        when(process.getProcessType()).thenReturn(WorkflowProcess.PROCESS_TYPE);
+        when(process.getName()).thenReturn("test-process");
+        when(process.getVersion()).thenReturn(VERSION);
+
+        when(processInstance.getContextInstance(eq(VariableScope.VARIABLE_SCOPE))).thenReturn(variableScope);
+        when(variableScope.getVariables()).thenReturn(processMetadata);
+
+        when(correlationKey.toExternalForm()).thenReturn("1");
+    }
+
+    /**
+     * Test build the ProcessInstanceLog for a regular process start
+     */
+    @Test
+    public void testBuildProcessStartedEvent() {
+        ServicesAwareAuditEventBuilder builder = new ServicesAwareAuditEventBuilder();
+        builder.setIdentityProvider(identityProvider);
+
+        ProcessStartedEvent pse = new ProcessStartedEventImpl(processInstance, kieRuntime);
+        ProcessInstanceLog log = (ProcessInstanceLog) builder.buildEvent(pse);
+
+        assertEquals("testUser", log.getIdentity());
+    }
+
+    /**
+     * Test build the ProcessInstanceLog for a process with initiator metadata
+     * and user auth bypasss not enabled
+     */
+    @Test
+    public void testBuildProcessStartedEventWithInitiatorAndNoUserAuthBypass() {
+
+        processMetadata.put("initiator", "john");
+
+        ServicesAwareAuditEventBuilder builder = new ServicesAwareAuditEventBuilder();
+        builder.setIdentityProvider(identityProvider);
+
+        ProcessStartedEvent pse = new ProcessStartedEventImpl(processInstance, kieRuntime);
+        ProcessInstanceLog log = (ProcessInstanceLog) builder.buildEvent(pse);
+
+        assertEquals("testUser", log.getIdentity());
+    }
+
+    /**
+     * Test build the ProcessInstanceLog for a process with initiator metadata
+     * and user auth bypasss enabled
+     */
+    @Test
+    public void testBuildProcessStartedEventWithInitiatorAndUserAuthBypassEnabled() {
+
+        processMetadata.put("initiator", "john");
+
+        ServicesAwareAuditEventBuilder builder = new ServicesAwareAuditEventBuilder();
+        builder.setIdentityProvider(identityProvider);
+
+        enableSetInitiator(builder);
+
+        ProcessStartedEvent pse = new ProcessStartedEventImpl(processInstance, kieRuntime);
+        ProcessInstanceLog log = (ProcessInstanceLog) builder.buildEvent(pse);
+
+        assertEquals("john", log.getIdentity());
+    }
```

Comment:
What about the 4th possible scenario - bypass is true but there is no initiator? Will the identity be null?

Reply:
No, it will use the data from the identity provider.
But I'll make a test for it.

Reply:
Test done

Reply:
Sure, I thought that if you bypass auth then the identity provider will return nothing, but of course there has to be something set, it will just not use the auth for engine operations...